pkg/netpol/eval/internal/k8s/representative_selectors.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package k8s
+
+import (
+	"crypto/sha1" //nolint:gosec // Non-crypto use
+	"encoding/hex"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// this file handles checking equality of representative peers' selectors' requirements
+// since a representative peer is a peer inferred from a network-policy rule
+
+// areRequirementsEqual given two labelSelector objects, one from a policy rule and the other from a represenatative pod/namespace,
+// returns true if:
+// 1. both selectors point to same reference (rule and its matching representative pod/ns)
+// 2. if the rule's selector is empty (matches all pods/namespaces)
+// 3. if the requirements of both rules are equal (same)
+func areRequirementsEqual(ruleSelector, repSelector *v1.LabelSelector) (bool, error) {
+	if ruleSelector == repSelector { // both label selectors point to same reference
+		return true, nil
+	}
+	// otherwise, check requirements equality
+	selector1, err := v1.LabelSelectorAsSelector(ruleSelector)
+	if err != nil {
+		return false, err
+	}
+	if selector1.Empty() { // empty rule matches everything
+		return true, nil
+	}
+	selector2, err := v1.LabelSelectorAsSelector(repSelector)
+	if err != nil {
+		return false, err
+	}
+	requirements1, _ := selector1.Requirements() // Requirements() returns sorted by key list
```

Comment:
where is it documented that `Requirements()` returns a sorted list?

Reply:
I found this by reading the implementation.
`LabelSelectorAsSelector` builds a new `labels.selector` by using `Requirements.Add` which sorts the requirements slice by key. 
and `Requirements()` returns this list 
links:
[Add](https://github.com/kubernetes/apimachinery/blob/d7e1c5311169d5ece2db0ae0118066859aa6f7d8/pkg/labels/selector.go#L373)
[LabelSelectorAsSelector](https://github.com/kubernetes/apimachinery/blob/dc7e034c86479d49be4b0eefad307621e10caa0e/pkg/apis/meta/v1/helpers.go#L34)

Reply:
- please add links in the comment about the expected sorted result. 
- please add unit tests to the functions in this file.

Reply:
> * please add links in the comment about the expected sorted result.
 done

Reply:
>     * please add unit tests to the functions in this file.

done